squarescale/squarescale-cli
# [138] Refactor environment variables display

:warning: This needs to be merged in conjunction with squarescale/squarescale-qa#24, after a QA run.

This PR introduces a well defined struct to hold the environment variables instead of a struct using string maps.
It also adds a filtering function taking options as a parameter to separate the filtering part from the command execution.

A new test suite covers both the changes to the Environment struct and the filtering methods. A QA test will cover the command usage.

## Pre submit Checklist

To be filled by the author.

### General

- [x] Higher level issue: squarescale/platform#138
- [x] In the scope of the current sprint

### Type of change

- [ ] :sparkles: New Feature
- [x] :art: Refactoring
- [ ] :shirt: Removing lint warnings
- [ ] :memo: Documentation
- [x] :white_check_mark: Adding tests
- [ ] :green_heart: Fixing CI
- [ ] :blue_heart: Fixing QA
- [ ] :yellow_heart: Fixing Rollbar
- [ ] :arrow_up: Upgrading dependencies
- [ ] :arrow_down: Downgrading dependencies

### Scope of change

- [x] :lipstick: Frontend
- [ ] :wrench: Back-end, API
- [ ] :rocket: Deployement, configuration
- [ ] :lock: Security
- [ ] :card_index: Data migration

### How to test

- [ ] Test scenario (please describe it)
> 1. Create a project with a DB
> 2. Add a service
> 3. Run `sqsc env get -project <project name>` 
> ⇒ check the DB_NAME value for the service is not the same as the one for the project
> 4. Run `sqsc env get -project <project name> -container <service name>`
> ⇒ check the DB_NAME value for the service is not the same as the one for the project
> 5. Run `sqsc env get -project <project name> -container <service name> DB_NAME`
> ⇒ check the value is not the same as the one for the project
- [x] Fully covered by unit tests
- [ ] Fully covered by QA tests
> A PR to come will add a QA test that covers the issue

### Misc

- [x] Commit messages are clear, log is readable
- [x] Branch name and commit messages contain reference to the higher issue
- [x] Code is rebased on top of master
- [x] Request is added to the task board

## Review Checklist

To be filled by the reviewers.

- [ ] No Hound violation
- [ ] Code coverage is ok
- [ ] Goal and impact of the feature are clear
